table_foosball/table_foosball_01.rb

```diff
@@ -0,0 +1,141 @@
+require "securerandom"
+
+class Player
+  def initialize name
+    @name = name
+    @games = [] # what are all the games this player has been involved in?
+    
+    # IDEAS for attributes
+    # - Time stamp for creation of this player object
+    # - UUID because multiple players might have similar names
+    @timestamp = Time.now
+    @id = ID.get_id
+  end
+
+  attr_reader :name, :games, :timestamp, :id
+
+  # return all the games this player won in
+  def won_games
+    @games.select {|g| g.winner.include? self}
+  end
+
+  # return all the games this player lost in
+  def lost_games
+    @games.select {|g| g.loser.include? self}
+  end
+  
+  # return all the games this player has played against opponent o
+  def no_of_games_against o
```

Comment:
Instead I would define `games_against(opponent)`. If you ever want the length, you can call `#length` on the result yourself.